test/org/sagebionetworks/bridge/dynamodb/DynamoReportDataDaoTest.java

```diff
@@ -0,0 +1,101 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static org.junit.Assert.assertEquals;
+import static org.sagebionetworks.bridge.TestConstants.TEST_STUDY;
+
+import org.joda.time.LocalDate;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.models.DateRangeResourceList;
+import org.sagebionetworks.bridge.models.reports.ReportData;
+import org.sagebionetworks.bridge.models.reports.ReportDataKey;
+import org.sagebionetworks.bridge.models.reports.ReportType;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+@ContextConfiguration("classpath:test-context.xml")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class DynamoReportDataDaoTest {
+    
+    private static final LocalDate END_DATE = LocalDate.parse("2016-03-31");
+
+    private static final LocalDate START_DATE = LocalDate.parse("2016-03-29");
+    
+    @Autowired
+    DynamoReportDataDao dao;
+    
+    private String reportId;
+    
+    private ReportDataKey reportDataKey;
+    
+    @Before
+    public void before() {
+        reportId = TestUtils.randomName(DynamoReportDataDaoTest.class);
+        reportDataKey = new ReportDataKey.Builder().withReportType(ReportType.STUDY).withIdentifier(reportId)
+                .withStudyIdentifier(TEST_STUDY).build();
+    }
+    
+    @After
+    public void after() {
+        if (reportDataKey != null) {
+            dao.deleteReportData(reportDataKey);    
+        }
+    }
+    
+    @Test
+    public void canCrud() {
+        ReportData report1 = createReport(LocalDate.parse("2016-03-30"), "a", "b");
+        ReportData report2 = createReport(LocalDate.parse("2016-03-31"), "c", "d");
+        
+        // These are not in date order. They should come back in date order.
+        dao.saveReportData(report2);
+        dao.saveReportData(report1);
+        
+        DateRangeResourceList<? extends ReportData> results = dao.getReportData(
+                reportDataKey, START_DATE, END_DATE);
```

Comment:
Is it worth adding a report outside of the date range so make sure the date range query works? What about a report with a different key?